packages/miniflare/test/plugins/core/inspector-proxy/index.spec.ts

```diff
@@ -0,0 +1,440 @@
+
+		const res = await resPromise;
+		t.is(await res.text(), "worker-a -> worker-b");
+	}
+);
+
+// The runtime inspector can send messages larger than 1MB limit websocket message permitted by UserWorkers.
+// In the real-world, this is encountered when debugging large source files (source maps)
+// or inspecting a variable that serializes to a large string.
+// Connecting devtools directly to the inspector would work fine, but we proxy the inspector messages
+// through the InspectorProxy, we need to make sure that such proxying does not hit the limit.
+// By logging a large string we can verify that the inspector messages are being proxied successfully.
+// (This issue was encountered with the wrangler inspector proxy worker: https://github.com/cloudflare/workers-sdk/issues/5297)
+test.serial("InspectorProxy: can proxy messages > 1MB", async (t) => {
+	const LARGE_STRING = "This is a large string => " + "z".repeat(2 ** 20);
+
+	const originalConsoleLog = console.log;
+	console.log = () => {};
+
+	const mf = new Miniflare({
+		inspectorPort: 9212,
+		unsafeInspectorProxy: true,
+		workers: [
+			{
+				script: `
+						export default {
+							fetch(request, env, ctx) {
+								console.log("${LARGE_STRING}");
+								return new Response(\`body:${LARGE_STRING}\`);
+							}
+						}
+					`,
+				modules: true,
+				unsafeInspectorProxy: true,
+			},
+		],
+	});
+	t.teardown(() => mf.dispose());
+
+	await mf.ready;
+
+	// when the inspector proxy is created there are some ws message exchanges between it
+	// and the runtime, these can interfere with testing and make tests flaky, so we just
+	// wait a bit here to avoid such problems
+	await setTimeout(30);
+
+	// Connect inspector WebSocket
+	const ws = new WebSocket("ws://localhost:9212");
+	const messages = events.on(ws, "message");
+	async function nextMessage() {
+		const messageEvent = (await messages.next()).value;
+		return JSON.parse(messageEvent[0].toString());
+	}
+
+	await events.once(ws, "open");
+
+	ws.send(JSON.stringify({ id: 0, method: "Debugger.enable" }));
+
+	t.like(await nextMessage(), {
+		method: "Debugger.scriptParsed",
+	});
+
+	t.like(await nextMessage(), { id: 0 });
+
+	// Send request and hit `debugger;` statement
+	const originalAssertConsumed = process.env.MINIFLARE_ASSERT_BODIES_CONSUMED;
+	process.env.MINIFLARE_ASSERT_BODIES_CONSUMED = undefined;
+	const resPromise = mf.dispatchFetch("http://localhost");
+	process.env.MINIFLARE_ASSERT_BODIES_CONSUMED = originalAssertConsumed;
+
+	const msg: {
+		method: string;
+		params: {
+			type: string;
+			args: { type: string; value: unknown }[];
+		};
+	} = await nextMessage();
+
+	t.is(msg.method, "Runtime.consoleAPICalled");
+	t.is(msg.params.type, "log");
+	t.is(msg.params.args.length, 1);
+	t.is(msg.params.args[0].type, "string");
+	t.is(msg.params.args[0].value, LARGE_STRING);
+
+	const res = await resPromise;
+	t.is(await res.text(), `body:${LARGE_STRING}`);
+
+	console.log = originalConsoleLog;
```

Comment:
Should be in a teardown?

```suggestion
	t.teardown(() => {
		console.log = originalConsoleLog;
	});
```

Reply:
of course! 😓 🤦